util/b58-payloads/src/payloads.rs

```diff
@@ -71,11 +76,15 @@ fn decode_payload(
     Ok((version, buffer_bytes.to_vec()))
 }
 
-/// Validate a fog_url
-/// TODO: improve test
-fn validate_fog_url(fog_url: &str) -> Result<(), Error> {
-    if fog_url.len() > 255 {
-        return Err(Error::TooManyBytes("fog_url".to_owned()));
+/// Validate a fog_report_url
+fn validate_fog_report_url(fog_report_url: &str) -> Result<(), Error> {
+    if fog_report_url.len() > 255 {
+        return Err(Error::TooManyBytes("fog_report_url".to_owned()));
+    }
+    // Empty fog url is okay, otherwise it must parse as FogUri
+    // TODO: Reject uri's containing a path?
+    if !fog_report_url.is_empty() {
+        FogUri::from_str(fog_report_url).map_err(Error::FogUrlParsingError)?;
```

Comment:
@rjwalters @kylefleming @voloshyn this may cause old payloads not to be valid anymore, if the fog urls dont have `fog://` or `insecure-fog://` scheme